src/adapters/mc/OsConfigResource.c

```diff
@@ -378,6 +384,108 @@ static MI_Result SetDesiredObjectValueToDevice(const char* who, char* objectName
     return miResult;
 }
 
+static MI_Result SetProcedureObjectValueToDevice(const char* who, char* objectName, MI_Context* context)
+{
+    char* payloadString = NULL;
+    int payloadSize = 0;
+
+    JSON_Value* jsonValue = NULL;
+    char* serializedValue = NULL;
+
+    MI_Result miResult = MI_RESULT_OK;
+    int mpiResult = MPI_OK;
+
+    if ((NULL == objectName) || (NULL == g_procedureObjectValue))
+    {
+        LogError(context, miResult, GetLog(), "[%s] SetProcedureObjectValueToDevice called with an invalid object name and/or procedure object value", who);
+        return MI_RESULT_INVALID_PARAMETER;
+    }
+
+    if (NULL == (jsonValue = json_value_init_string(g_procedureObjectValue)))
+    {
+        miResult = MI_RESULT_FAILED;
+        mpiResult = ENOMEM;
+        LogError(context, miResult, GetLog(), "[%s] json_value_init_string('%s') failed", who, g_procedureObjectValue);
+    }
+    else if (NULL == (serializedValue = json_serialize_to_string(jsonValue)))
+    {
+        miResult = MI_RESULT_FAILED;
+        mpiResult = ENOMEM;
+        LogError(context, miResult, GetLog(), "[%s] json_serialize_to_string('%s') failed", who, g_procedureObjectValue);
+    }
+    else
+    {
+        payloadSize = (int)strlen(serializedValue);
+        if (NULL != (payloadString = malloc(payloadSize + 1)))
+        {
+            const OsConfigComponent *component = NULL;
```

Comment:
>const OsConfigComponent *component  [](http://example.com/codeflow?start=12&length=35)

Pure C style requires all variables locally declared withing a statement of a function to be defined at the beginning/top of the statement. To respect this old style and for consistency with existing code, can you please move the definition on line 421 at top, between lines 389 and 396? Please repeat the same for all other locals that would fall into this case.